### DISCUSSION TITLE
App URI query strings are stripped

### DISCUSSION THREAD

**Discussion by gprinsen** [gravitational/teleport#12051] [Q&A]
It appears that when an App URI contains a query string, that it's stripped from the URI that is forwarded to the origin server. For example, the following in an app_service:

`uri: http://localhost:80/site?page=1`

Results in the URI `http://localhost:80/site` being requested and returned.

Am I missing something, or is this by design? And if it's by design, is there a URL rewrite function that allows the requested URI to be re-written to achieve what my application requires?

Thanks

**Comment by webvictim** [comment]
You might try quoting the URI to make sure it isn't a YAML parsing issue, i.e. `uri: "http://localhost:80/site?page=1"`

**Reply by gprinsen** [reply]
Thanks, yes already tried that however no luck. One of the examples in the documentation even shows a question mark without any special need to escape it, so it may be a bug, however I note that there are some special use-cases for kubes for which the example applies.
